helm/botkube/templates/communicationsecret.yaml

```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "botkube.fullname" . }}-communication-secret
+  labels:
+    app.kubernetes.io/name: {{ include "botkube.name" . }}
+    helm.sh/chart: {{ include "botkube.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+stringData:
+  comm_config.yaml: |
+    # Communication settings
+    communications:
+      {{- with .Values.communications }}
+        {{- toYaml . | nindent 4 }}
+      {{- end }}
```

Comment:
@slalwani97 Unfortunately we had to revert the commit. The problem is with the indentation here. The secret gets created with following format. That's why it couldn't get unmarshalled into struct 
```
# Communication settings
communications:
elasticsearch:
  enabled: false
  index:
    name: botkube
    replicas: 0
    shards: 1
    type: botkube-event
  password: ELASTICSEARCH_PASSWORD
  server: ELASTICSEARCH_ADDRESS
  username: ELASTICSEARCH_USERNAME
mattermost:
  channel: dev-alerts
  enabled: false
  notiftype: short
  team: MATTERMOST_TEAM
  token: xxxx
  url: xxxx
slack:
  channel: general
  enabled: true
  notiftype: short
  token: xxxxxxxx
webhook:
  enabled: false
  url: WEBHOOK_URL
```

Reply:
Probably changing nindent to 6 will solve the problem.

Reply:
```
/config $ cat  comm_config.yaml 
# Communication settings
communications:
  elasticsearch:
    enabled: false
    index:
      name: botkube
      replicas: 0
      shards: 1
      type: botkube-event
    password: ELASTICSEARCH_PASSWORD
    server: ELASTICSEARCH_ADDRESS
    username: ELASTICSEARCH_USERNAME
  mattermost:
    channel: MATTERMOST_CHANNEL
    enabled: false
    notiftype: short
    team: MATTERMOST_TEAM
    token: MATTERMOST_TOKEN
    url: MATTERMOST_SERVER_URL
  slack:
    channel: SLACK_CHANNEL
    enabled: false
    notiftype: short
    token: SLACK_API_TOKEN
  webhook:
    enabled: false
    url: WEBHOOK_URL
```